pydantic/main.py

```diff
@@ -453,7 +467,8 @@ def validate(cls: Type['Model'], value: Any) -> 'Model':
             return cls.from_orm(value)
         else:
             with change_exception(DictError, TypeError, ValueError):
-                return cls(**dict(value))
+                value_dict = dict(value)
+            return cls(**value_dict)
```

Comment:
Note that since `ValidationError` is a subclass of `ValueError`, the old behavior was actually causing failure to parse as cls to be (confusingly) raised as a `DictError`. This has actually been a frequent source of confusion for new FastAPI users (related to `orm_mode` being off), but until now I didn't understand why it was happening.

I think the change I made here is "correct" (it didn't cause any errors in the existing tests), but let me know if I'm overlooking something.

(This change was necessary to get a reasonable error message in the case where you try to call `parse_as_type` and one of the field values is incompatible with the superclass type for that field.)

Reply:
I think this is fine, might be a good idea regardless of this PR.

Reply:
This may be worth a PR prior to v1 since it changes the validation error type in some cases. Feel free to just make the two line PR for me if I can’t get to it before the v1.0 beta.

Reply:
agreed @dmontagu, you could create a PR for this together with test that demonstrates the change?